src/dotnet-install.sh

```diff
@@ -816,39 +847,76 @@ install_dotnet() {
     zip_path="$(mktemp "$temporary_file_template")"
     say_verbose "Zip path: $zip_path"
 
-    say "Downloading link: $download_link"
 
     # Failures are normal in the non-legacy case for ultimately legacy downloads.
     # Do not output to stderr, since output to stderr is considered an error.
+    say "Downloading link: $download_link"
+
+    # the download function will fill variables $http_code and $download_error_msg in case of failure.
+    http_code=""; download_error_msg=""
     download "$download_link" "$zip_path" 2>&1 || download_failed=true
+    primary_path_http_code="$http_code"; primary_path_download_error_msg="$download_error_msg"
 
     #  if the download fails, download the legacy_download_link
     if [ "$download_failed" = true ]; then
-        say "Cannot download: $download_link"
+        case $primary_path_http_code in
+        404)
+            say "The primary path $download_link is not available."
+            ;;
+        2*)
+            say "Cannot download via the primary path $download_link."
```

Comment:
Can we reuse `primary_path_download_error_msg` here?  Seems logic is duplicated

Reply:
Yes, I do not like ending up with duplicated code too. Messages are quite similar, but not the same. I may try to use a single message in these two cases and it will give me an opportunity to reuse, but that will mean I either skip part of the information in messages (which link I now download, primary or legacy), or I provide information that users do not need to see if everything goes fine (Is it wget or curl download).

Reply:
Well, I just got an idea, if I put the information about the link being privacy or legacy in the message
```
say "Downloading link: $download_link"
``` 
it will allow me to skip this information when I write the error message and remove this duplication. I suppose I will go for that.